components/automate-ui/src/app/modules/desktop/dashboard/dashboard.component.ts

```diff
@@ -193,26 +222,97 @@ export class DashboardComponent implements OnInit {
   }
 
-  public onErrorSelected(errorItem: TopErrorsItem): void {
+  public onDailyStatusSelected(status: string, last24HourCheckInCount: DailyCheckInCount) {
+    if (!status) { return this.onDesktopListClose(); }
+
+    let start: Date;
+    let end: Date;
+
+    switch (status) {
+      case ('checked-in'):
+        start = new Date(last24HourCheckInCount.start);
+        end = new Date(last24HourCheckInCount.end);
+        break;
+      case ('unknown'):
+        end = new Date(last24HourCheckInCount.start);
+        break;
+    }
+
+    this.store.dispatch(new UpdateDesktopListTitle(`Daily Check-in: ${status}`));
+    this.store.dispatch(new UpdateDesktopDateTerm({ start, end }));
+    this.store.dispatch(new UpdateDesktopFilterTerm({ terms: [] }));
+    this.store.dispatch(new GetNodeMetadataCounts());
+    this.store.dispatch(new GetDesktops());
+    this.selectedCheckInStatus = status;
+    this.selectedError = undefined;
+    this.selectedDuration = undefined;
+    this.desktopListVisible = true;
+  }
+
+  public onTermFilterAdded(term: TermFilter) {
+    this.store.dispatch(new AddDesktopFilterTerm({ term }));
+    this.store.dispatch(new GetNodeMetadataCounts());
+    this.store.dispatch(new GetDesktops());
+  }
+
+  public onTermFilterRemoved(term: TermFilter) {
+    this.store.dispatch(new RemoveDesktopFilterTerm({ term }));
+    this.store.dispatch(new GetNodeMetadataCounts());
+    this.store.dispatch(new GetDesktops());
+  }
+
+  public onErrorSelected(error: TopErrorsItem): void {
+    if (!error) { return this.onDesktopListClose(); }
+
     const terms = [
-      { type: Terms.ErrorMessage, value: errorItem.message },
-      { type: Terms.ErrorType, value: errorItem.type }];
+      { type: Terms.ErrorType, value: error.type },
+      { type: Terms.ErrorMessage, value: error.message },
+      { type: Terms.Status, value: 'failure' }
+    ];
+    this.store.dispatch(new UpdateDesktopListTitle(`${error.type}: ${error.message}`));
+    this.store.dispatch(new UpdateDesktopDateTerm({}));
     this.store.dispatch(new UpdateDesktopFilterTerm({ terms }));
+    this.store.dispatch(new GetNodeMetadataCounts());
+    this.store.dispatch(new GetDesktops());
+    this.selectedError = error;
+    this.selectedDuration = undefined;
+    this.selectedCheckInStatus = '';
     this.desktopListVisible = true;
   }
 
-  public onDurationSelected(_durationItem: any): void {
-    this.store.dispatch(new UpdateDesktopFilterCurrentPage({ page: 1 }));
+  public onDurationSelected(duration: string): void {
+    if (!duration) { return this.onDesktopListClose(); }
+
+    const units = { 'M': 'month', 'w': 'week', 'd': 'day' };
+    const unit = units[duration[1]];
+    const amount = parseInt(duration[0], 10);
+    const end = moment().utc().subtract(amount, unit).toDate();
+    this.store.dispatch(new UpdateDesktopListTitle(`Last Check-in: ${amount} ${unit}${amount > 1 ? 's' : ''} ago`));
```

Comment:
Presumably zero could occur as well, and this will account for it:
```suggestion
    this.store.dispatch(new UpdateDesktopListTitle(`Last Check-in: ${amount} ${unit}${amount != 1 ? 's' : ''} ago`));
```